nilearn/mass_univariate/_utils.py

```diff
@@ -140,6 +130,15 @@ def calculate_tfce(
     return tfce_4d
 
 
+def _return_score_threshs(arr3d, dh, two_sided_test):
+    """Compute list of score threshold to use for TFCE."""
+    max_score = (
+        np.nanmax(np.abs(arr3d)) if two_sided_test else np.nanmax(arr3d)
+    )
+    number_steps = 100 if dh == "auto" else max(10, round(max_score / dh))
```

Comment:
```suggestion
    number_steps = 100 if dh == "auto" else max(10, round(max_score / dh))
    if number_steps < 10:
       warns(f'Not enough steps for TFCE. Got: {number_steps=}. '
       'Setting it to 10.', stacklevel=3
       )
       number_steps = 10 
```

Reply:
Thank you.  Q: we have `max(10, round(max_score / dh))` in there so the warning will never be issued with the code the way it is.  If we change that to simply `round(max_score / dh)` then the warning will be issued when `round(max_score / dh) < 10`.  Despite that these two possible implementations have different behaviors with the warning, with both we proceed with `number_steps = 10` whenever `round(max_score / dh)` is too small, so that's good.  Do we want to trigger the warning when `round(max_score / dh) < 10` by changing `max(10, round(max_score / dh))` to `round(max_score / dh)`?

At the risk of feature creep ... do we also want to warn and do something sensible when `number_steps` is too large?  How about when `np.isnan(number_steps)`?

Reply:
I just pushed a commit that checks that `number_steps` does not exceed 1000.  The code warns when the value has to be coerced to be within the range of 10 to 1000.

Reply:
> CI informs me that `warns` is not defined, and exploration indicates that it is for pytest code rather than somewhere like `_utils.py`. I have just pushed a commit that clips to the range of 10 to 1000, but does not issue a warning.

ah yeah you probably need to to from `warnings import warn` to get it from the standard library